wp-content/civi-extensions/goonjcustom/Civi/InductionService.php

```diff
@@ -69,108 +73,171 @@ public static function volunteerCreated(string $op, string $objectName, int $obj
     \Civi::log()->info('Volunteer set: ', [
       'id' => self::$volunteerId,
     ]);
-
   }
 
   /**
-   *
+   * Common logic for creating an induction.
    */
-  public static function createInductionForVolunteer(string $op, string $objectName, int $objectId, &$objectRef) {
-    if ($op !== 'create' || $objectName !== 'Address') {
+  private static function createInduction(int $contactId, int $stateId) {
+    if (self::inductionExists($contactId)) {
+      \Civi::log()->info('Induction already exists for contact', ['id' => $contactId]);
       return FALSE;
     }
 
-    \Civi::log()->info('Address created: ', [
-      'id' => $objectId,
-      'volunteer' => self::$volunteerId,
-      'contactId' => $objectRef->contact_id,
-      'isPrimary' => $objectRef->is_primary,
+    if (!$stateId) {
+      return FALSE;
+    }
+
+    $office = self::findOfficeForState($stateId);
+    if (!$office) {
+      return FALSE;
+    }
+
+    $coordinatorId = self::findCoordinatorForOffice($office['id']);
+    if (!$coordinatorId) {
+      return FALSE;
+    }
+
+    $sourceContactId = self::getCurrentUserOrVolunteer($contactId);
+    $targetContactId = ($sourceContactId === $contactId) ? $contactId : $contactId;
+
+    $placeholderActivityDate = self::getPlaceholderActivityDate();
+
+    \Civi::log()->info('Before induction activity create: ', [
+      'source' => $sourceContactId,
+      'target' => $targetContactId,
+      'coordinator' => $coordinatorId,
+      'officeId' => $office['id'],
     ]);
 
-    if (self::$volunteerId !== $objectRef->contact_id || !$objectRef->is_primary) {
+    Activity::create(FALSE)
+      ->addValue('activity_type_id:name', self::INDUCTION_ACTIVITY_TYPE_NAME)
+      ->addValue('status_id:name', self::INDUCTION_DEFAULT_STATUS_NAME)
+      ->addValue('source_contact_id', $sourceContactId)
+      ->addValue('target_contact_id', $targetContactId)
+      ->addValue('Induction_Fields.Assign', $coordinatorId)
+      ->addValue('activity_date_time', $placeholderActivityDate)
+      ->addValue('Induction_Fields.Goonj_Office', $office['id'])
+      ->execute();
+
+    return TRUE;
+  }
+
+  /**
+   * Handles induction creation for a volunteer.
+   */
+  public static function createInductionForVolunteer(string $op, string $objectName, int $objectId, &$objectRef) {
+    if ($op !== 'create' || $objectName !== 'Address' || self::$volunteerId !== $objectRef->contact_id || !$objectRef->is_primary) {
       return FALSE;
     }
 
     $stateId = $objectRef->state_province_id;
 
-    $officesFound = Contact::get(FALSE)
+    if (!$stateId) {
+      \Civi::log()->info('state not found', ['VolunteerId' => self::$volunteerId]);
```

Comment:
I think you need a return statement after this

Reply:
sure added